### PR TITLE
Support deposit receipts in the YAML file

### LINKED ISSUE #29: Add expected deposit receipts field to yaml files
EEs could output deposit receipts (via `eth2_pushNewDeposit` method). The expected receipt (if any) should be added to the YAML file to make it testable.

Related to #3

### PR DESCRIPTION
**Pull request by g-r-a-n-t**
Closes #29 

This PR includes the following changes:
- Scout will now collect deposit receipts outputted at each block during execution.
- Scout will read a list of deposit receipts from the yaml file.
- Scout will compare the receipts within the yaml file against what was outputted during execution. If a receipt that is provided within the yaml file is not outputted during execution, it will exit with 1. If a receipt is outputted during execution that is not provided by the yaml file, it will continue and not exit with 1.
- A minimal script based off of helloworld has been added as a sanity check.

### CODE REVIEW

**Comment by g-r-a-n-t** [comment]
Hey @s1na, this may be of interest to you.

File: src/main.rs [hunk 1]
```diff
@@ -725,12 +757,36 @@ fn process_yaml_test(filename: &str) {
...
+        println!("No deposit receipts to check.")
+    } else if test_deposit_receipts
+        .iter()
+        .all(|r| deposit_receipts.contains(r))
+    {
+        println!("All test deposit receipts were created.")
+    } else {
+        println!("Missing deposit receipts");
+        println!("Expected to see: {:?}", test_deposit_receipts);
+        println!("Found: {:?}", deposit_receipts);
```
Comment (axic) [inline src/main.rs]: I'd change these to:
```
println!("Expected deposit receipts: {}", expected_deposit_receipts);
println!("Got deposit receipts: {}", deposit_receipts);
```

File: src/main.rs [hunk 2]
```diff
@@ -725,12 +757,36 @@ fn process_yaml_test(filename: &str) {
...
-            .expect("processing shard block to succeed")
+        deposit_receipts.append(
+            process_shard_block(&mut shard_state, &beacon_state, Some(block.into()))
+                .expect("processing shard block to succeed")
+                .as_mut(),
+        );
+    }
+
+    if test_deposit_receipts.is_empty() {
+        println!("No deposit receipts to check.")
```
Comment (axic) [inline src/main.rs]: I'd remove this case.

File: scripts/deposit/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    let pre_state_root = eth2::load_pre_state_root();
+
+    assert!(eth2::block_data_size() == 0);
+
+    // No updates were made to the state
+    let post_state_root = pre_state_root;
+
+    // This is valid length-wise, but invalid content-wise
+    let deposit = [0u8; 184];
+    eth2::push_new_deposit(&deposit);
```
Comment (axic) [inline scripts/deposit/src/lib.rs]: This example script doesn't seem to differ from helloworld. I'd remove it.

File: src/main.rs [hunk 4]
```diff
@@ -725,12 +757,36 @@ fn process_yaml_test(filename: &str) {
     let beacon_state: BeaconState = test_file.beacon_state.into();
     let pre_state: ShardState = test_file.shard_pre_state.into();
     let post_state: ShardState = test_file.shard_post_state.into();
+    let test_deposit_receipts: Vec<Deposit> = test_file
+        .deposit_receipts
+        .into_iter()
+        .map(|r| r.into())
```
Comment (axic) [inline src/main.rs]:
```suggestion
        .map(|receipt| receipt.into())
```

**Comment by axic** [comment]
@g-r-a-n-t thank you for this work!

**Comment by g-r-a-n-t** [comment]
Thanks for the review @axic. All comments have been addressed.

**Comment by axic** [comment]
Thanks @g-r-a-n-t and sorry for the delay!

**Review by axic — APPROVED** [review]

**Pull request merged**